conn.go

```diff
@@ -851,6 +873,19 @@ func (c *Conn) serverHandshake() error {
 		}
 	}
 
+	// Find early_data extension and handle early data
+	sendALPN := false
+	var serverALPN *alpnExtension
+	if gotALPN {
+		for _, proto := range clientALPN.protocols {
+			if c.config.enabledProto[proto] {
```

Comment:
I would tend to prefer searching through the list here rather than building an unnecessary map.